lib/fastlane/plugin/wpmreleasetoolkit/actions/common/upload_build_to_apps_cdn.rb

```diff
@@ -0,0 +1,322 @@
+# frozen_string_literal: true
+
+require 'fastlane/action'
+require 'net/http'
+require 'uri'
+require 'json'
+
+module Fastlane
+  module Actions
+    module SharedValues
+      APPS_CDN_UPLOADED_FILE_URL = :APPS_CDN_UPLOADED_FILE_URL
+      APPS_CDN_UPLOADED_FILE_ID = :APPS_CDN_UPLOADED_FILE_ID
+      APPS_CDN_UPLOADED_POST_ID = :APPS_CDN_UPLOADED_POST_ID
+      APPS_CDN_UPLOADED_POST_URL = :APPS_CDN_UPLOADED_POST_URL
+    end
+
+    class UploadBuildToAppsCdnAction < Action
+      RESOURCE_TYPE = 'Build'
+      VALID_POST_STATUS = %w[publish draft].freeze
+      VALID_BUILD_TYPES = %w[Alpha Beta Nightly Production Prototype].freeze
+      VALID_PLATFORMS = ['Android', 'iOS', 'Mac - Silicon', 'Mac - Intel', 'Mac - Any', 'Windows'].freeze
+
+      def self.run(params)
+        UI.message('Uploading build to apps CDN...')
+
+        file_path = params[:file_path]
+        UI.user_error!("File not found at path '#{file_path}'") unless File.exist?(file_path)
+
+        api_endpoint = "https://public-api.wordpress.com/rest/v1.1/sites/#{params[:site_id]}/media/new"
+        uri = URI.parse(api_endpoint)
+
+        # Create the request body and headers
+        parameters = {
+          product: params[:product],
+          build_type: params[:build_type],
+          visibility: params[:visibility].to_s.capitalize,
+          platform: params[:platform],
+          resource_type: RESOURCE_TYPE,
```

Comment:
I can clearly see this constant being set as a parameter, however I'm getting:

```
Uploading build to apps CDN...
[15:41:32]: Failed to upload build to apps CDN: 422
[15:41:32]: {"errors":[{"file":"rabbit.zip","error":"validation_error","message":"The resource_type field must be one of: Build."}]}

[!] Upload to apps CDN failed
```

When running the command from the "Manual Testing" section while supplying:
- The proper API token
- An updated version, build number, and release notes
- An updated file path to the file I'm trying to upload

Reply:
Nice catch, this was due to a small refactoring I did after my initial testing, that ended up introducing an extra newline at the end of each `form-data` parameter… and I forgot to re-test afterwards.

Fixed in a283fc1.